src/yunohost/user.py

```diff
@@ -462,6 +469,276 @@ def user_info(auth, username):
+    }
+
+    if auth.add('cn=%s,ou=groups' % groupname, attr_dict):
+        logger.success(m18n.n('group_created'))
+        if sync_perm:
+            permission_sync_to_user(auth)
+        return {'name': groupname}
+
+    raise YunohostError('group_creation_failed')
+
+
+@is_unit_operation([('groupname', 'user')])
+def user_group_delete(operation_logger, auth, groupname, force=False, sync_perm=True):
+    """
+    Delete user
+
+    Keyword argument:
+        groupname -- Groupname to delete
+
+    """
+    from yunohost.permission import permission_sync_to_user
+
+    if not force and (groupname == 'all_users' or groupname == 'admins' or groupname in user_list(auth, ['uid'])['users']):
+        raise YunohostError('group_deletion_not_allowed', user=groupname)
+
+    operation_logger.start()
+    if not auth.remove('cn=%s,ou=groups' % groupname):
+        raise YunohostError('group_deletion_failed')
+
+    logger.success(m18n.n('group_deleted'))
+    if sync_perm:
+        permission_sync_to_user(auth)
+
+
+@is_unit_operation([('groupname', 'user')])
+def user_group_update(operation_logger, auth, groupname, add_user=None, remove_user=None, force=False, sync_perm=True):
+    """
+    Update user informations
+
+    Keyword argument:
+        groupname -- Groupname to update
+        add_user -- User to add in group
+        remove_user -- User to remove in group
+
+    """
+
+    from yunohost.permission import permission_sync_to_user
+
+    attrs_to_fetch = ['member']
+
+    if (groupname == 'all_users' or groupname == 'admins') and not force:
+        raise YunohostError('edit_group_not_allowed', group=groupname)
+
+    # Populate group informations
+    result = auth.search(base='ou=groups,dc=yunohost,dc=org',
+                         filter='cn=' + groupname, attrs=attrs_to_fetch)
+    if not result:
+        raise YunohostError('group_unknown', group=groupname)
+    group = result[0]
+
+    new_group_list = {'member': set(), 'memberUid': set()}
+    if 'member' in group:
+        new_group_list['member'] = set(group['member'])
+    else:
+        group['member'] = []
+
+    user_l = user_list(auth, ['uid'])['users']
+
+    if add_user:
+        if not isinstance(add_user, list):
+            add_user = [add_user]
+        for user in add_user:
+            if not user in user_l:
+                raise YunohostError('user_unknown', user=user)
+            userDN = "uid=" + user + ",ou=users,dc=yunohost,dc=org"
+            if userDN in group['member']:
+                logger.warning(m18n.n('user_already_in_group', user=user, group=groupname))
+            new_group_list['member'].add(userDN)
+
+    if remove_user:
+        if not isinstance(remove_user, list):
+            remove_user = [remove_user]
+        for user in remove_user:
+            userDN = "uid=" + user + ",ou=users,dc=yunohost,dc=org"
+            if user == groupname:
+                raise YunohostError('remove_user_of_group_not_allowed', user=user, group=groupname)
```

Comment:
same as for the previous block : let's have do this check independently of starting to actually do stuff ?